### PR TITLE
Validate numerical scheme RibaMetaMod

### PR DESCRIPTION
**Pull request by HendrikKok**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -134,3 +134,5 @@ report.xml
 
 .pixi
 .imod_collector
+.teamcity
+LHM2016_v01vrz
```
Comment (Huite) [inline .gitignore:138]: Are these needed?

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 2]
```diff
@@ -137,6 +137,9 @@ def initialize(self) -> None:
             self.ribasim.initialize(ribasim_config_file)
         if self.has_metaswap:
```
Comment (Huite) [inline imod_coupler/drivers/ribametamod/ribametamod.py:138]: The `couple` function is rather long now.

Could you rewrite it slightly, so that it essentially reads:

```python
def couple(self):
        self.max_iter = self.mf6.max_iter()
        mf6_flowmodel_key = self.coupling.mf6_model
        self.mf6_head = self.mf6.get_head(mf6_flowmodel_key)

        if self.has_ribasim:
            self.couple_ribasim()

        if self.has_metaswap:
            self.couple_metaswap()
```

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 3]
```diff
@@ -303,17 +310,18 @@ def update(self) -> None:
 
         self.mf6.prepare_time_step(0.0)
         self.delt_gw = self.mf6.get_time_step()
-        self.subtimesteps_sw = range(1, int(self.delt_gw / self.delt_sw) + 1)
 
         if self.has_ribasim:
+            self.ribasim.update_subgrid_level()
```
Comment (Huite) [inline imod_coupler/drivers/ribametamod/ribametamod.py:315]: Move this into a separate method called `update_ribasim_modflow()` or `exchange_ribasim_modflow()`.

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 4]
```diff
@@ -303,17 +310,18 @@ def update(self) -> None:
...
             # zeros exchange-arrays, Ribasim pointers and API-packages
             self.exchange.reset()
             # exchange stage and compute flux estimates over MODFLOW 6 timestep
             self.exchange_stage_rib2mod()
             self.exchange.add_flux_estimate_mod(self.delt_gw, self.mf6_head)
 
         if self.has_metaswap and self.has_ribasim:
```
Comment (Huite) [inline imod_coupler/drivers/ribametamod/ribametamod.py:322]: Similarly here, into `update_ribasim_metaswap` or `exchange_ribasim_metaswap`.

**Review by Huite — CHANGES_REQUESTED** [review]
Looks good, just split a few of the longer methods into submethods.

File: .gitignore (hunk 1 above)
Comment (rleander) [reply .gitignore:138]: wel als je de unsa database in je git working copy directory heb staan, maar het is de vraag of dat handig is. Ik heb m daarbuiten en ook de imod_collector. maar waarvoor heb je .teamcity nodig eigenlijk?

File: imod_coupler/drivers/ribametamod/config.py [hunk 5]
```diff
@@ -21,8 +21,8 @@ class Coupling(BaseModel):
     mf6_passive_drainage_packages: dict[str, str]
 
     enable_sprinkling_groundwater: bool = False  # true when sprinkling is active
-    mf6_msw_recharge_pkg: str | None = (
-        None  # the recharge package that will be used for coupling
+    mf6_msw_recharge_pkg: str = (
+        ""  # the recharge package that will be used for coupling
```
Comment (Huite) [inline imod_coupler/drivers/ribametamod/config.py:25]: Why is this required? I prefer None instead of an empy string to be honest.

File: imod_coupler/drivers/ribametamod/exchange.py [hunk 6]
```diff
@@ -194,7 +199,7 @@ def to_modflow(self, realised: NDArray[np.float64]) -> None:
             # in which case the Modflow demand was POSITIVE, otherwise the correction is 0
             self.mf6_active_river_api_packages[key].rhs[:] = -(
                 np.maximum(self.demands_mf6[key], 0.0)
-                * (1 - self.mapping.mod2rib[key].transpose().dot(realised_fraction))
+                * (1 - self.mapping.map_mod2rib[key].transpose().dot(realised_fraction))
```
Comment (Huite) [inline imod_coupler/drivers/ribametamod/exchange.py:202]: I think it's better to do this transpose once, then just re-use it.
Generally a transpose results in new allocations etc.

**Review by Huite — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (HendrikKok) [reply .gitignore]: Removed both

File: imod_coupler/drivers/ribametamod/config.py (hunk 5 above)
Comment (HendrikKok) [reply imod_coupler/drivers/ribametamod/config.py:25]: indeed
